File: dbxWearablesApp/dbxWearablesApp/Models/SyncRecord.swift
import Foundation

/// A record of a single sync POST to the Databricks endpoint.
struct SyncRecord: Codable, Identifiable {
    let id: UUID
    let recordType: String
    let timestamp: Date
    let recordCount: Int
    let httpStatusCode: Int
    let success: Bool
    /// Raw NDJSON string. Only populated for last-payload storage; nil in recent events log.
    let ndjsonPayload: String?
    let requestHeaders: [String: String]

    enum CodingKeys: String, CodingKey {
        case id
        case recordType = "record_type"
        case timestamp
        case recordCount = "record_count"
        case httpStatusCode = "http_status_code"
        case success
        case ndjsonPayload = "ndjson_payload"
        case requestHeaders = "request_headers"
    }
}

File: androidHealthConnect/app/src/main/java/com/dbxwearables/android/data/model/SyncRecord.kt
package com.dbxwearables.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SyncRecord(
    val id: String,
    @SerialName("record_type") val recordType: String,
    val timestamp: String,
    @SerialName("record_count") val recordCount: Int,
    @SerialName("http_status_code") val httpStatusCode: Int,
    val success: Boolean,
    @SerialName("ndjson_payload") val ndjsonPayload: String? = null,
    @SerialName("request_headers") val requestHeaders: Map<String, String> = emptyMap()
)
